ddp-workspace/projects/ddp-dsm-ui/src/app/scanner/scanner.component.ts

```diff
@@ -67,24 +69,24 @@ export class ScannerComponent implements DoCheck, OnDestroy {
 
   public removeFields(index: number): void {
     this.scannerFields.length - 1 && this.scannerFields.removeAt(index);
+    this.shortIds.length - 1 && this.shortIds.splice(index, 1);
     this.activeScanner.inputFields.forEach((inputField: InputField) =>
       this.checkForDuplicates(inputField.controllerName));
   }
 
   public save(): void {
     this.additionalMessage = '';
     const filteredActiveScannerFieldsGroupsArray = this.filteredNonNullFieldsGroups;
-
+    this.shortIds = Array(...Array(this.scannerFields.length)).map(function() {});
```

Comment:
I don't think we need to do that, as we can still add items to the array, even without explicitly defining its length in forward (I mean at any index).

**Example:**
```
const exampleArray = [];
exampleArray[2] = 'data 1'
exampleArray[5] = 'data 2'

console.log(exampleArray)
console.log(exampleArray[2])
console.log(exampleArray[5])

// Output:
// [empty × 2, 'data 1', empty × 2, 'data 2']
// data 1
// data 2
```

Or if you still want to use the Array constructor, you could simplify your code like so:
`this.shortIds = Array(this.scannerFields.length)`

Reply:
Since if they remove the entry at an index, the short id is also removed at that index, then we need to initialize the short id.  Otherwise you are correct